subprojects/core-api/src/main/java/org/gradle/api/artifacts/FileCollectionDependency.java

```diff
@@ -28,4 +32,29 @@ public interface FileCollectionDependency extends SelfResolvingDependency {
      * @since 3.3
      */
     FileCollection getFiles();
+
+    /**
+     * {@inheritDoc}
+     *
+     * @deprecated This class will no longer implement {@link SelfResolvingDependency} in Gradle 9.0.
+     */
+    @Override
+    @Deprecated
+    TaskDependency getBuildDependencies();
+
+    /**
+     * {@inheritDoc}
+     *
+     * @deprecated This class will no longer implement {@link SelfResolvingDependency} in Gradle 9.0. Use {@link #getFiles()} instead.
+     */
+    @Override
+    Set<File> resolve();
```

Comment:
💅 Shouldn't these be marked `@Deprecated` as well, to be consistent with `getBuildDependencies()`?  Applies throughout.